activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/JsonSchemaFlattener.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.organization.validation;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.springframework.core.io.ClassPathResource;
+    
+public class JsonSchemaFlattener {
+    
+    private Map<String, Object> addDefinitions = new HashMap<>();
+    
+    public JsonSchemaFlattener() {
+         
+    }
+    
+    private Optional<JSONObject> handleJSONObject(JSONObject jsonObject) {
+        
+        Optional<JSONObject> reply = Optional.empty();
+        
+        if (!jsonObject.isEmpty()) {
+            Iterator iterator = jsonObject.keys();
+            
+            while (iterator.hasNext()) {
+                String key = (String) iterator.next();
+                Object value = jsonObject.get(key);
+                
+                if (isKeyToCheck(key)) {
+                    Optional<String> updatedString = getUpdatedValue(value);    
+                    if (updatedString.isPresent()) {
+                        jsonObject.put(key, updatedString.get());
+                        
+                        reply = Optional.of(jsonObject);
+                    }
+                    
+                } else {                  
+                    if (value instanceof JSONObject) {
```

Comment:
Block from line 61 to 76 is duplicated with a similar block (lines 90 to 104). This can be avoided if you extract a method handling the value only. I'll try to provide an example.

Reply:
Agree

Reply:
@eromano @erdemedeiros 
A big refactoring made  for `JsonSchemaFormatter`

- avoid using a single char variable name (refactor your code accordingly)
Done

- there is a catch without any code inside
Such cases are avoided now

- often you return null. We should avoid propagating null
No returns of NULL, use Optional now

Duplicated codes are removed and everything simplified

Map with definitions is initialized locally now and used as a parameter by functions call
`Map<String, Object> addDefinitions = new HashMap<>()`
So no problems with concurrent calls should be possible in principle.